src/Razor/src/Microsoft.AspNetCore.Razor.VSCode.Extension/syntaxes/aspnetcorerazor.tmLanguage.yml

```diff
@@ -434,7 +435,7 @@ repository:
 
   using-directive:
     name: 'meta.directive.using.cshtml'
-    match: '(@)(using)\b\s*(?!\()(.+?)?(;)?$'
+    match: '(@)(using)\b\s*(?!\(|\s)(.+?)?(;)?$'
```

Comment:
Maybe I'm reading this wrong, but this regex seems too permisive (and kind of weird). If I try to parse it as natural language I figure it's something like: Capture group 0 is an '@', followed by capture group 1 which is 'using', followed by a word boundary and some possible white-space, then a Negative Lookahead for either '(' or some whitespace, a word, then a possible ';' and a possible end of line. According to my regex tester that would allow for strings like `@using sdfsdff;)hjkgh` to be considered matches?

Reply:
Yup, that's valid. Razor will parse it, send it to C# and then C# will complain 😄